client/src/modules/functions/functions.js

```diff
@@ -0,0 +1,98 @@
+angular.module('bhima.controllers')
+.controller('FunctionManagementController', FunctionManagementController);
+
+FunctionManagementController.$inject = [
+  'FunctionService', 'ModalService',
+  'NotifyService', 'uiGridConstants', '$state', 'SessionService',
+];
+
+/**
+ * Function Management Controller
+ *
+ * This controller is about the Job Title management module in the admin zone
+ * It's responsible for creating, editing and updating a Job Title
+ */
+function FunctionManagementController(Functions, ModalService,
+  Notify, uiGridConstants, $state, Session) {
+  var vm = this;
+
+  // bind methods
+  vm.deleteFunction = deleteFunction;
+  vm.editFunction = editFunction;
+  vm.createFunction = createFunction;
+  vm.toggleFilter = toggleFilter;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      { field : 'fonction_txt', displayName : 'FORM.LABELS.DESIGNATION', headerCellFilter : 'translate' },
+      { field : 'action',
+        width : 80,
+        displayName : '',
+        cellTemplate : '/modules/functions/templates/action.tmpl.html',
+        enableSorting : false,
+        enableFiltering : false,
+      },
+    ],
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  }
+
+  function loadFunctions() {
+    vm.loading = true;
+
+    Functions.read(null, { detailed : 1 })
+    .then(function (data) {
+      vm.gridOptions.data = data;
+    })
+    .catch(Notify.handleError)
+    .finally(function () {
+      vm.loading = false;
+    });
+  }
+
+  // switch to delete warning mode
+  function deleteFunction(title) {
```

Comment:
Instead of "title" which sounds like it might be a `String`, try calling this `profession`.  I thought it was a text string at first.

... alternatively, just pass in the `id`.  It looks like all you actually need is the `id`.